src/hooks/monitors.ts

```diff
@@ -8,17 +8,26 @@ type UseParams = {
   monitor_id?: string;
 };
 
-export function useMonitor(params: UseParams, queryOptions?: UseQueryOptions<MonitorTypes, Error>) {
+export function useMonitor(
+  params: UseParams,
+  queryOptions?: UseQueryOptions<MonitorTypes, Error, MonitorTypes>
+) {
   const fetchMonitors = () =>
     API.request({
       method: 'GET',
       url: '/monitors',
-      params: {
-        ...params,
-      },
+      params,
       ...queryOptions,
-    }).then((response: AxiosResponse<{ monitors: MonitorTypes[] }>) => response.data.monitors[0]);
-  return useQuery(['monitors', params], fetchMonitors, {
+    }).then((response: AxiosResponse<{ monitors: MonitorTypes[] }>) =>
+      response.data.monitors.find((monitor) => monitor.id === params.monitor_id)
+    );
+  return useQuery(['monitor', params], fetchMonitors, {
+    placeholderData: {
+      id: '',
+      title: '',
+      description: '',
+      geostories: [],
+    },
```

Comment:
I think you are not using the placeholder correctly, to me we don't really have this need. This is only when you want to show something before the data is loaded, or as default data.